### PR TITLE
test for pub_handler

### PR DESCRIPTION
**Pull request by Hermann0222**

### CODE REVIEW

File: .github/workflows/coverage.yml [hunk 1]
```diff
@@ -26,6 +26,6 @@ jobs:
       run: cd build && ctest --output-on-failure
 
     - name: Upload report
-      uses: codecov/codecov-action@v1
+      uses: codecov/codecov-action@v3
       with:
```
Comment (wanghaEMQ) [inline .github/workflows/coverage.yml:30]: Hi. what's changed in version 3?

File: nanomq/pub_handler_test.c [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+	nng_msg_alloc(&msg, 0);
+	pub_packet                         = nng_zalloc(sizeof(*pub_packet));
+	work->msg                          = msg;
+	work->pub_packet                   = pub_packet;
+	int                  remaining_len = 18;
+	struct fixed_header *fix_hd        = nng_alloc(sizeof(*fix_hd));
+	fix_hd->qos                        = 0;
+	fix_hd->packet_type                = PUBLISH;
+	int topic_len                      = 7;
+	int data_len                       = 4;
```
Comment (wanghaEMQ) [inline nanomq/pub_handler_test.c:28]: Did you forget to format it? Or make it clear

**Review by wanghaEMQ — CHANGES_REQUESTED** [review]
Hi. I am so sorry. Your codes in pub_handler_test are hard to read. Could you make it clear or add more comments? plz

File: .github/workflows/coverage.yml (hunk 1 above)
Comment (Hermann0222) [reply .github/workflows/coverage.yml:30]: As of February 1, 2022, v1 has been fully sunset and no longer functions

File: nanomq/pub_handler_test.c (hunk 2 above)
Comment (Hermann0222) [reply nanomq/pub_handler_test.c]: I will format it.

File: .github/workflows/coverage.yml (hunk 1 above)
Comment (wanghaEMQ) [reply .github/workflows/coverage.yml:30]: Okey. Good!

File: nanomq/pub_handler_test.c [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+	remaining_len = 11;
+	nng_msg_set_remaining_len(work->msg, remaining_len);
+	rv_rc = decode_pub_message(work, MQTT_PROTOCOL_VERSION_v311);
+	assert(rv_rc == SUCCESS);
+	// check work->pub_packet
+	assert(work->pub_packet->var_header.publish.topic_name.len == 5);
+	assert(strcmp(work->pub_packet->var_header.publish.topic_name.body,
+	           "$MQTT") == 0);
+	assert(work->pub_packet->payload.len == 4);
+	assert(strcmp(pub_packet->payload.data, "data") == 0);
```
Comment (wanghaEMQ) [inline nanomq/pub_handler_test.c:64]: Payload is formatted as binary. It's better to compare bytes in payload one by one.

File: nanomq/pub_handler_test.c [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+	nng_msg_set_cmd_type(dest_msg, CMD_PUBLISH);
+	rv_bool = encode_pub_message(dest_msg, work, PUBLISH);
+	assert(rv_bool == true);
+
+	// check dest_msg
+	// args to get the content of msg
+	uint32_t pos            = 0;
+	uint32_t dest_topic_len = 0;
+	uint8_t *dest_body      = nng_msg_body(msg);
+	// check topic len
```
Comment (wanghaEMQ) [inline nanomq/pub_handler_test.c:79]: A test for a topic without right length should be added.

**Review by wanghaEMQ — APPROVED** [review]

**Review by wanghaEMQ — APPROVED** [review]

**Pull request merged**